src/lib/format/l20n/ast/parser.js

```diff
@@ -5,25 +5,32 @@ import { L10nError } from '../../../errors';
 
 const MAX_PLACEABLES = 100;
 
-export default {
-  parse: function(env, string, pos = false) {
-    this._source = string;
-    this._index = 0;
-    this._length = string.length;
-    this._curEntryStart = 0;
+
+class ParseContext {
+  constructor(env, pos) {
     this._config = {
       pos: pos
     };
+  } 
 
-    if (pos !== true) {
-      AST.Node.prototype.setPosition = function() {};
+  setPosition(node, start, end) {
+    if (!this._config.pos) {
+      return;
     }
-    return this.getResource(pos);
-  },
+    node._pos = {start, end};
+  }
+
+  parseResource(string) {
+    this._source = string;
+    this._index = 0;
+    this._length = string.length;
+    this._curEntryStart = 0;
```

Comment:
I think these should be set in the constructor?  This way it will be possible to create a new `ParseContext` and call _any_ method.  This is what #62 might want to use, for instance, with `getEntry`.